### PR TITLE
Add include-what-you-use script and pragmas

### PR DESCRIPTION
**Pull request by sethrj**
This PR is for setting up a future one-click source update so it only has manual changes. @drbenmorgan I did have to add a bunch of mappings for the Geant4 `.icc` classes but it works after the fact, so I think the approach we discussed before (a script that generates an `.imp` file with the `icc->h` remapping based on spidering the icc files) should work.

### CODE REVIEW

File: src/corecel/Assert.hh [hunk 1]
```diff
@@ -286,10 +287,11 @@
...
 #else
-#    define CELER_HIP_CALL(STATEMENT)    \
-        do                               \
-        {                                \
-            CELER_NOT_CONFIGURED("HIP"); \
+#    define CELER_HIP_CALL(STATEMENT)                      \
+        do                                                 \
+        {                                                  \
+            CELER_NOT_CONFIGURED("HIP");                   \
+            (void)sizeof(celeritas_device_runtime_api_h_); \
```
Comment (amandalund) [inline src/corecel/Assert.hh:295]: Is this not needed for `CELER_CUDA_CALL`?

File: src/corecel/device_runtime_api.h [hunk 2]
```diff
@@ -49,4 +49,12 @@
 #    define CELER_EU_PER_CU 0
 #endif
 
+/*!
+ * Declare a dummy variable to be referenced in disabled "CELER_BLAH" calls.
+ *
+ * With this declaration, the build will fail if this include isn't missing
```
Comment (amandalund) [inline src/corecel/device_runtime_api.h:55]: If it is missing?

**Review by amandalund — APPROVED** [review]
Looks good to me @sethrj!

File: src/corecel/Assert.hh (hunk 1 above)
Comment (sethrj) [reply src/corecel/Assert.hh:295]: Not usually, as NVCC I think implicitly includes this? Though I guess we could be calling this from a cc file too so maybe I'll add it

**Pull request merged**